rasa/graph_components/validators/config_validator.py

```diff
@@ -0,0 +1,532 @@
+from __future__ import annotations
+from collections import defaultdict
+import itertools
+from typing import Iterable, List, Dict, Text, Any, Set, Type
+
+from rasa.engine.graph import ExecutionContext, GraphComponent, GraphSchema, SchemaNode
+from rasa.engine.storage.storage import ModelStorage
+from rasa.engine.storage.resource import Resource
+from rasa.nlu.featurizers.featurizer import Featurizer2
+from rasa.nlu.extractors.mitie_entity_extractor import (
+    MitieEntityExtractorGraphComponent,
+)
+from rasa.nlu.extractors.regex_entity_extractor import (
+    RegexEntityExtractorGraphComponent,
+)
+from rasa.nlu.extractors.crf_entity_extractor import CRFEntityExtractorGraphComponent
+from rasa.nlu.extractors.entity_synonyms import EntitySynonymMapperComponent
+from rasa.nlu.featurizers.sparse_featurizer.regex_featurizer import (
+    RegexFeaturizerGraphComponent,
+)
+from rasa.nlu.classifiers.diet_classifier import DIETClassifierGraphComponent
+from rasa.nlu.selectors.response_selector import ResponseSelectorGraphComponent
+from rasa.nlu.tokenizers.tokenizer import TokenizerGraphComponent
+from rasa.core.policies.rule_policy import RulePolicyGraphComponent
+from rasa.core.policies.policy import PolicyGraphComponent, SupportedData
+from rasa.core.policies.memoization import MemoizationPolicyGraphComponent
+from rasa.core.policies.ted_policy import TEDPolicyGraphComponent
+from rasa.nlu.featurizers.sparse_featurizer.lexical_syntactic_featurizer import (
+    LexicalSyntacticFeaturizerGraphComponent,
+)
+from rasa.shared.core.training_data.structures import RuleStep, StoryGraph
+from rasa.shared.constants import (
+    DEFAULT_CONFIG_PATH,
+    DOCS_URL_COMPONENTS,
+    DOCS_URL_DEFAULT_ACTIONS,
+    DOCS_URL_POLICIES,
+    DOCS_URL_RULES,
+)
+from rasa.shared.core.domain import Domain, InvalidDomain
+from rasa.shared.core.constants import (
+    ACTION_BACK_NAME,
+    ACTION_RESTART_NAME,
+    USER_INTENT_BACK,
+    USER_INTENT_RESTART,
+)
+from rasa.shared.exceptions import InvalidConfigException
+from rasa.shared.importers.importer import TrainingDataImporter
+from rasa.shared.nlu.training_data.training_data import TrainingData
+import rasa.shared.utils.io
+
+
+# FIXME: check right paths, not just existence on any path (e.g. end2end featurization
+# might be separate from NLU featurization and only one of them might have a
+# tokenizer)
+
+TRAINABLE_EXTRACTORS = {
```

Comment:
You can get exactly the same information from the `DefaultV1Recipe` [here](https://github.com/RasaHQ/rasa/blob/8f1469a1fd913ae165e536a91c06e35cfd0e112f/rasa/engine/recipes/default_recipe.py#L200)

Reply:
ah, perfect :)

Reply:
It's not in main yet - shall we just leave this PR open until that recipe is merged so I can reuse that information?

Reply:
How merging it like it is and then I/you put up another PR which integrate them on the integration branch?